### ISSUE TITLE
Able to set IOPS with "knife vsphere vm vmdk add"?

### ISSUE THREAD

**Issue by petericebear** [chef-partners/knife-vsphere#448]
Hi,

Is it possible to set the IOPS of a new vmdk on attachment to a VM?
We have SSD, Block and Archive which we want to limit IOPS based on the datastore it goed to.

Thanks in advance, for reply

**Comment by swalberg** [comment]
We don't expose that directly. Would need to look up where in the API this gets set. _might_ be able to do it through `knife vsphere vm config`...

Which version of vsphere is this for?

**Comment by petericebear** [comment]
Do you have an idea how to set this, at which level?

**Comment by swalberg** [comment]
I couldn't find anything about iops. There was some stuff about disk shares (SharesInfo) but I'm not clear if that translates into iops. I think the easiest way to find it is to set it in the web UI and then try and find it in a console session. Just haven't had the time this week -- my day just just changed into our busy period.

**Comment by petericebear** [comment]
It should be possible: 
https://github.com/lamw/vghetto-scripts/blob/master/powershell/configurePerVMDKIOPS.ps1

In this PowerShell script the setting can be set. I will look into this, and will give you an update.

**Comment by petericebear** [comment]
https://gist.github.com/petericebear/02739f8b9b3c4dd20040b6b996d7e722

**Comment by swalberg** [comment]
Aha, good find. Same comment as the other one -- did you want to PR that after switching to SearchHelpers and ui.info?

Or let me know if you want me to do it; just wanted to make sure you have the opportunity for the credit in the commit log.

**Comment by petericebear** [comment]
Yeah will do, think at the end of the week I have some time for that
